FirebaseVertexAI/Sources/Types/Internal/Imagen/DecodableImagenImage.swift

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+protocol DecodableImagenImage: ImagenImage, Decodable {
+  init(mimeType: String, bytesBase64Encoded: String?, gcsURI: String?)
```

Comment:
Is it possible to have both? Otherwise should it be an enum to choose between the two?

Like https://github.com/firebase/firebase-ios-sdk/blob/main/FirebaseAuth/Sources/Swift/AuthProvider/EmailAuthProvider.swift#L45

Reply:
@paulb777  How strongly do you feel about this refactor? It's not possible to have both but, although the `DecodableImagenImage` protocol is internal, `ImagenImage` will be part of the public API. I want to avoid adding enums to the public API since they are semver breaking to add or remove cases.

I'm also not sure that `DecodableImagenImage` (or `InternalImagenImage`) will make it into the final product so I'm reluctant to spend more time editing them. They will likely be replaced by specific types for inline data and GCS images.

Reply:
Fine to leave it as is.